internal/web_api.go

```diff
@@ -112,14 +104,13 @@ func (a *API) UploadFile(filename string, file []byte) (string, error) {
 	if token == nil {
 		return "", fmt.Errorf("couldn't get token")
 	}
-	url := fmt.Sprintf("%s/v%s/%s/action/upload_file", a.APIURL, a.version, a.name)
-	if token.LicenseID != nil {
-		url = fmt.Sprintf("%s?license_id=%v", url, *token.LicenseID)
-	}
-	req, err := http.NewRequest("POST", url, body)
+
+	req, err := a.requestGetter(token, "upload_file")
 	if err != nil {
-		return "", fmt.Errorf("couldn't create new POST request to '%v': %v", url, err)
+		return "", fmt.Errorf("couldn't create new POST request: %v", err)
 	}
+	req.Method = "POST"
```

Comment:
I think this error is misleading because you are setting request type after checking this error